bftengine/tests/testSerialization/TestSerialization.cpp

```diff
@@ -425,45 +425,49 @@ void testSetSimpleParams(bool toSet) {
 }
 
 void fillReplicaConfig() {
-  config.fVal = fVal;
-  config.cVal = cVal;
-  config.replicaId = 3;
-  config.numOfClientProxies = 8;
-  config.statusReportTimerMillisec = 15;
-  config.concurrencyLevel = 5;
-  config.autoViewChangeEnabled = true;
-  config.viewChangeTimerMillisec = 12;
-  config.maxExternalMessageSize = 2048;
-  config.maxNumOfReservedPages = 256;
-  config.maxReplyMessageSize = 1024;
-  config.sizeOfReservedPage = 2048;
-  config.debugStatisticsEnabled = true;
-
-  config.replicaPrivateKey = replicaPrivateKey;
-  config.publicKeysOfReplicas.insert(IdToKeyPair(0, publicKeyValue1));
-  config.publicKeysOfReplicas.insert(IdToKeyPair(1, publicKeyValue2));
-  config.publicKeysOfReplicas.insert(IdToKeyPair(2, publicKeyValue3));
-  config.publicKeysOfReplicas.insert(IdToKeyPair(3, publicKeyValue4));
-
-  config.thresholdSignerForExecution = nullptr;
-  config.thresholdVerifierForExecution = new IThresholdVerifierDummy;
-  config.thresholdSignerForSlowPathCommit = new IThresholdSignerDummy;
-  config.thresholdVerifierForSlowPathCommit = new IThresholdVerifierDummy;
-  config.thresholdSignerForCommit = new IThresholdSignerDummy;
-  config.thresholdVerifierForCommit = new IThresholdVerifierDummy;
-  config.thresholdSignerForOptimisticCommit = new IThresholdSignerDummy;
-  config.thresholdVerifierForOptimisticCommit = new IThresholdVerifierDummy;
+  config = new ReplicaConfig;
+  config->fVal = fVal;
+  config->cVal = cVal;
+  config->replicaId = 3;
+  config->numOfClientProxies = 8;
+  config->statusReportTimerMillisec = 15;
+  config->concurrencyLevel = 5;
+  config->autoViewChangeEnabled = true;
+  config->viewChangeTimerMillisec = 12;
+  config->maxExternalMessageSize = 2048;
+  config->maxNumOfReservedPages = 256;
+  config->maxReplyMessageSize = 1024;
+  config->sizeOfReservedPage = 2048;
+  config->debugStatisticsEnabled = true;
+
+  config->replicaPrivateKey = replicaPrivateKey;
+  config->publicKeysOfReplicas.insert(IdToKeyPair(0, publicKeyValue1));
+  config->publicKeysOfReplicas.insert(IdToKeyPair(1, publicKeyValue2));
+  config->publicKeysOfReplicas.insert(IdToKeyPair(2, publicKeyValue3));
+  config->publicKeysOfReplicas.insert(IdToKeyPair(3, publicKeyValue4));
+
+  config->thresholdSignerForExecution = nullptr;
+  config->thresholdVerifierForExecution = new IThresholdVerifierDummy;
+  config->thresholdSignerForSlowPathCommit = new IThresholdSignerDummy;
+  config->thresholdVerifierForSlowPathCommit = new IThresholdVerifierDummy;
+  config->thresholdSignerForCommit = new IThresholdSignerDummy;
+  config->thresholdVerifierForCommit = new IThresholdVerifierDummy;
+  config->thresholdSignerForOptimisticCommit = new IThresholdSignerDummy;
+  config->thresholdVerifierForOptimisticCommit = new IThresholdVerifierDummy;
+  config->singletonFromThis();
+  config->singletonFromThis();
```

Comment:
Why do this twice?

Reply:
just to check that it works as expected (only once)